system/pci_mmio.c

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "boot.h"
+#include "bootparams.h"
+
+#include "cpuid.h"
+#include "io.h"
+
+#include "mmio.h"
+
+#include "pci.h"
+#include "unistd.h"
+
+//------------------------------------------------------------------------------
+// Constants
+//------------------------------------------------------------------------------
+
+#define NB_PCI_MMIO_TYPE0_BASE 0x0EFE00000000
+#define NB_PCI_MMIO_TYPE1_BASE 0x0EFE10000000
+
+//------------------------------------------------------------------------------
+// Public Functions
+//------------------------------------------------------------------------------
+
+void pci_init(void)
+{
+    return;
+}
+
+uint64_t pci_config_type0_addr(int bus, int dev, int func, int reg)
+{
+    uint64_t addr = NB_PCI_MMIO_TYPE0_BASE
+                  | (reg  & 0xf00) << 16
+                  | (bus  & 0xff)  << 16
+                  | (dev  & 0x1f)  << 11
+                  | (func & 0x07)  << 8
+                  | (reg  & 0xff);
+    return (addr);
+}
+
+uint64_t pci_config_type1_addr(int bus, int dev, int func, int reg)
+{
+    uint64_t addr = NB_PCI_MMIO_TYPE1_BASE
+                  | (reg  & 0xf00) << 16
+                  | (bus  & 0xff)  << 16
+                  | (dev  & 0x1f)  << 11
+                  | (func & 0x07)  << 8
+                  | (reg  & 0xff);
+    return (addr);
+}
+
+uint8_t pci_config_read8(int bus, int dev, int func, int reg)
+{
+    if (bus == 0) {
+      return mmio_read8((uint8_t *)pci_config_type0_addr(bus, dev, func, reg));
```

Comment:
An occasional indent slip. You've globally done well, though :)

Reply:
OK, I will add more two spaces next time.